lxd/api_internal.go

```diff
@@ -135,7 +135,7 @@ func internalContainerOnStart(d *Daemon, r *http.Request) response.Response {
 		return response.SmartError(fmt.Errorf("Instance is not container type"))
 	}
 
-	c := inst.(*containerLXC)
+	c := inst.(instance.Container)
```

Comment:
As a generic API, we could have a `OnHook(name string, data interface{})` function in the instance interface and use that to route that kind of hooks.

What do you think?

Reply:
@stgraber yep that would make things more generic, although as right now we only pass in strings, I think I'd prefer to use `OnHook(name string, args map[string]string) error` as otherwise we'd likely end up having to define concrete struct types for each type of hook argument in the `instance` package so that both caller and callee can share the definition, at which point we're basically back to defining container specific types again.

Reply:
fair enough

Reply:
and since our input is env variable or command arguments, we likely won't have anything other than strings anyway